api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel.types;
+
+import static com.datastrato.gravitino.rel.types.Types.DecimalType.checkPrecisionScale;
+
+import com.google.common.base.Preconditions;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Objects;
+
+/** Used to represent a {@link Types.DecimalType} value in Gravitino. */
+public class Decimal {
```

Comment:
Why do we need a `Decimal` value in Gravitino, can you please add more comments?

Reply:
At first, I intended to use the type name directly as the method name, such as `long()`, `float()`, `double()`. However, as they are reserved keywords in Java, I had to resort to using `longLiteral()`, `floatLiteral()`, `doubleLiteral()`. 

If we want to use `decimalLiteral()` here, should the already defined methods like `time()`, `date()`, `string()` be changed to `xxxLiteral()`?